File: Triplets.java
import java.util.*;
class Triplets {

	public static void findTriplets(int[] arr, LinkedList<int []> ans) {
		Arrays.sort(arr);
		for(int i = 0; i < arr.length-2;) {
			if(arr[i+1] == arr[i]+1) {
				if(arr[i+2] == arr[i]+2) {
					int[] temp = {arr[i], arr[i+1], arr[i+2]}; 
					ans.add(temp);
					i++;
				}
				else
					i += 2;
			}
			else
				i++;
		}
	}

	public static void main(String[] args) {

		//demo test cases
		//int[] arr = {1,2,3,4,12,34,32,21,0,34,6,56,7,34,23,99,66,77,44,33,66,77,98,97};
		//int[] arr = {1,2,3,4,5,6,7,8,9};

		LinkedList<int[]> ans = new LinkedList<int[]>();		
		findTriplets(arr, ans);

		for(int[] str : ans) {
			System.out.println(Arrays.toString(str));
		}
	}
}

File: Triplets.py

def find_triplets(l):
	length = len(l)
	triplets_list = []
	l.sort()
	for i in range(0, length-2):
		if((l[i+1] == l[i]+1) and(l[i+2] == l[i]+2)):
			triplets_list.append([l[i], l[i+1], l[i+2]])
	print(triplets_list)

def main():
	#l = [1,2,3,4,12,34,32,21,0,34,6,56,7,34,23,99,66,77,44,33,66,77,98,97]
	l = [1,2,3,4,5,6,7,8,9]
	find_triplets(l)

main()





